data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraService.java

```diff
@@ -0,0 +1,361 @@
+     *
+     * @param configuration Input Parameter
+     * @param ts            Input Parameter
+     * @return String Builder
+     */
+    private StringBuilder createIssueFilterCriteria(JiraSourceConfig configuration, Instant ts) {
+
+        log.info("Creating issue filter criteria");
+        if (!CollectionUtils.isEmpty(JiraConfigHelper.getProjectKeyFilter(configuration))) {
+            validateProjectFilters(configuration);
+        }
+        StringBuilder jiraQl = new StringBuilder(UPDATED + GREATER_THAN_EQUALS + ts.toEpochMilli());
+        if (!CollectionUtils.isEmpty(JiraConfigHelper.getProjectKeyFilter(configuration))) {
+            jiraQl.append(PROJECT_IN).append(JiraConfigHelper.getProjectKeyFilter(configuration).stream()
+                            .collect(Collectors.joining(DELIMITER, PREFIX, SUFFIX)))
+                    .append(CLOSING_ROUND_BRACKET);
+        }
+        if (!CollectionUtils.isEmpty(JiraConfigHelper.getIssueTypeFilter(configuration))) {
+            jiraQl.append(ISSUE_TYPE_IN).append(JiraConfigHelper.getIssueTypeFilter(configuration).stream()
+                            .collect(Collectors.joining(DELIMITER, PREFIX, SUFFIX)))
+                    .append(CLOSING_ROUND_BRACKET);
+        }
+        if (!CollectionUtils.isEmpty(JiraConfigHelper.getIssueStatusFilter(configuration))) {
+            jiraQl.append(STATUS_IN).append(JiraConfigHelper.getIssueStatusFilter(configuration).stream()
+                            .collect(Collectors.joining(DELIMITER, PREFIX, SUFFIX)))
+                    .append(CLOSING_ROUND_BRACKET);
+        }
+        log.info("Created issue filter criteria JiraQl query: {}", jiraQl);
+        return jiraQl;
+    }
+
+    /**
+     * Method for Validating Project Filters.
+     *
+     * @param configuration Input Parameter
+     */
+    private void validateProjectFilters(JiraSourceConfig configuration) {
+        log.info("Validating project filters");
+        List<String> badFilters = new ArrayList<>();
+        Pattern regex = Pattern.compile("[^A-Z0-9]");
+        JiraConfigHelper.getProjectKeyFilter(configuration).forEach(projectFilter -> {
+            Matcher matcher = regex.matcher(projectFilter);
+            if (matcher.find() || projectFilter.length() <= 1 || projectFilter.length() > 10) {
+                badFilters.add(projectFilter);
+            }
+        });
+        if (!badFilters.isEmpty()) {
+            String filters = String.join("\"" + badFilters + "\"", ", ");
+            log.error("One or more invalid project keys found in filter configuration: {}", badFilters);
+            throw new BadRequestException(BAD_REQUEST_EXCEPTION
+                    + filters);
+        }
+    }
+
+    /**
+     * Method for creating Item Info.
+     *
+     * @param key      Input Parameter
+     * @param metadata Input Parameter
+     * @return Item Info
+     */
+    private ItemInfo createItemInfo(String key, Map<String, Object> metadata) {
```

Comment:
Rather than building these, use Jackson to serialize it.

```
return objectMapper.convertValue(metadata, JiraItemInfo.class);
```

Here is an example: https://github.com/opensearch-project/data-prepper/blob/5108528b70da24c4749c657d84aa783896e93e64/data-prepper-plugins/event-json-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/event_json/EventJsonOutputCodec.java#L77

Reply:
Some of the field values in the jira response are like `j_issue_key` if we want to auto convert then we need to create similar attributes in the class. We are only picking up 3 attributes and constructing this class and logic is located in one single method for now.